ModuleInterface: Always ignore export-as for printing references in private swiftinterfaces

xymus opened the pull request:
Always print the real module name for references in private swiftinterfaces, ignoring export-as declarations. Keep using the export-as name for the public swiftinterface only.

The flag `ModuleInterfaceExportAs` used to enable this behavior and we're removing it to make it the default.

rdar://115922907

xymus committed:
[ModuleInterface] Always ignore export-as in private swiftinterfaces

Always print the real module name for references in private
swiftinterfaces, ignoring export-as declarations. Keep using the
export-as name for the public swiftinterface only.

The flag `ModuleInterfaceExportAs` used to enable this behavior and
we're removing it to make it the default.

rdar://115922907

xymus commented:
@swift-ci Please smoke test

tshortli commented on include/swift/Basic/Features.def:
```diff
@@ -147,7 +147,6 @@ EXPERIMENTAL_FEATURE(OneWayClosureParameters, false)
 EXPERIMENTAL_FEATURE(TypeWitnessSystemInference, false)
 EXPERIMENTAL_FEATURE(LayoutPrespecialization, true)
 
-EXPERIMENTAL_FEATURE(ModuleInterfaceExportAs, true)
```
What happens to projects that build with `-enable-experimental-feature ModuleInterfaceExportAs` after removing this? Don't we need to leave the feature name around, even though we are enabling it by default?

xymus replied:
I've added a test to see the effect of keeping the flag. Apparently the driver and frontend silently ignore unknown experimental flags. It's a feature of these flags to avoid errors but I wonder why there's not even a warning or a remark. I'll have to take a deeper look at this.

xymus replied:
I asked Doug about this, we need to keep the experimental flags pretty much forever when they are used in the swiftinterfaces feature checks. I think there's no clear story on how to remove these flags when we can. He was on board about adding a remark for unknown feature flags.

xymus committed:
[ModuleInterface] Test that ModuleInterfaceExportAs doesn't raise errors

Passing the flags `-enable-experimental-feature ModuleInterfaceExportAs`
has no effect anymore, test that it's not reported as an error.

xymus commented:
@swift-ci Please smoke test
